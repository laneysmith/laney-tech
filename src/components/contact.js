import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Section from './section';
import List from './section/list';
import ListItem from './section/listItem';

const Contact = () => {
  const data = useStaticQuery(graphql`
    query ContactQuery {
      site {
        siteMetadata {
          social {
            github
            linkedin
            email
          }
        }
      }
    }
  `);
  const { github, linkedin, email } = data.site.siteMetadata.social;

  return (
    <Section id="contact-section" title="Contact Me">
      <List>
        <ListItem icon="ðŸ’¾">
          <a href={`https://github.com/${github}`}>github.com/{github}</a>
        </ListItem>
        <ListItem icon="ðŸ‘”">
          <a href={`https://linkedin.com/in/${linkedin}`}>linkedin.com/in/{linkedin}</a>
        </ListItem>
        <ListItem icon="ðŸ“¨">
          <a href="mailto:lsmith.unc@gmail.com">{email}</a>
        </ListItem>
      </List>
    </Section>
  );
};

export default Contact;
