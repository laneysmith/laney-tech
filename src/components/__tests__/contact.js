import React from 'react';
import { render as testRender } from '@testing-library/react';
import { PureContact as Contact } from '../contact';

describe('Contact', () => {
  it('should match snapshot', () => {
    const data = {
      site: {
        siteMetadata: {
          social: {
            github: 'my-github-username',
            linkedin: 'my-linkedin-username',
            email: 'me@theinternet.com',
          },
        },
      },
    };
    const { container } = testRender(<Contact data={data} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
