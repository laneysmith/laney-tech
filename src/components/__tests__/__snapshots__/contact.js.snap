// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
.c0 {
  margin: 0 0 0.8125rem 0;
  padding: 0 0.8125rem;
}

.c1 {
  padding: 0.8125rem 0 0.8125rem 0.8125rem;
  margin-left: 0.8125rem;
  border-left: 1px solid;
}

.c2 {
  margin: 0.5em 0 0.5em 0.5em;
  display: grid;
  grid-template-columns: 20px 1fr;
  grid-column-gap: 8px;
  grid-row-gap: 4px;
  grid-auto-flow: row;
}

.c2 > [role='img'] {
  font-size: 1.2em;
}

@media only screen and (max-width:600px) {
  .c0 {
    padding: 0;
  }
}

<section
  class="c0"
  id="contact-section"
>
  <h3>
    Contact Me
  </h3>
  <main
    class="c1"
  >
    <div
      class="c2"
      role="list"
    >
      <div
        role="img"
      >
        ğŸ’¾
      </div>
      <div
        role="listitem"
      >
        <a
          href="https://github.com/my-github-username"
        >
          github.com/
          my-github-username
        </a>
      </div>
      <div
        role="img"
      >
        ğŸ‘”
      </div>
      <div
        role="listitem"
      >
        <a
          href="https://linkedin.com/in/my-linkedin-username"
        >
          linkedin.com/in/
          my-linkedin-username
        </a>
      </div>
      <div
        role="img"
      >
        ğŸ“¨
      </div>
      <div
        role="listitem"
      >
        <a
          href="mailto:me@theinternet.com"
        >
          me@theinternet.com
        </a>
      </div>
    </div>
  </main>
</section>
`;
